abstractions/dotnet/src/store/InMemoryBackingStore.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Kiota.Abstractions.Store {
+    public class InMemoryBackingStore : IBackingStore {
+        private bool isInitializationComplete = true;
+        public bool ReturnOnlyChangedValues {get; set;}
+        private readonly Dictionary<string, Tuple<bool, object>> store = new();
+        private Dictionary<string, Action<string, object, object>> subscriptions = new();
+        public T Get<T>(string key) {
+            if(string.IsNullOrEmpty(key))
+                throw new ArgumentNullException(nameof(key));
+
+            if(store.TryGetValue(key, out var result))
+                return ReturnOnlyChangedValues && result.Item1 || !ReturnOnlyChangedValues ? (T)result.Item2 : default;
+            else
+                return default;
+        }
+        public void Set<T>(string key, T value) {
+            if(string.IsNullOrEmpty(key))
+                throw new ArgumentNullException(nameof(key));
+
+            var valueToAdd = new Tuple<bool, object>(InitializationCompleted, value);
+            Tuple<bool, object> oldValue = null;
+            if(!store.TryAdd(key, valueToAdd)) {
+                oldValue = store[key];
+                store[key] = valueToAdd;
+            }
+            foreach(var sub in subscriptions.Values)
+                sub.Invoke(key, oldValue?.Item2, value);
+        }
+        public IEnumerable<KeyValuePair<string, object>> Enumerate() {
+            return (ReturnOnlyChangedValues ? store.Where(x => !x.Value.Item1) : store)
+                .Select(x => new KeyValuePair<string, object>(x.Key, x.Value.Item2));
+        }
+        public string Subscribe(Action<string, object, object> callback) {
```

Comment:
why would they need to provide their id in your opinion?

Reply:
I imagine a chain of event driven changes starting from UI through different layers of applications using object handles to identify the change. If the consumer has their own identifier, then it can be propagated through to our object subscription. **One less identifier mapping required for them to manage is the main value I see.**

Reply:
makes sense, thanks for the feedback. Added.